memory/migrate_pages.py

```diff
@@ -50,16 +50,23 @@ def setUp(self):
             self.cancel('Test requires two numa nodes to run.'
                         'Node list with memory: %s' % nodes)
 
-        dist = distro.detect()
+        self.dist = distro.detect()
         pkgs = ['gcc', 'make']
-        if dist.name in ["Ubuntu", 'debian']:
+        if self.dist.name in ["Ubuntu", 'debian']:
             pkgs.extend(['libpthread-stubs0-dev',
                          'libnuma-dev', 'libhugetlbfs-dev'])
-        elif dist.name in ["centos", "rhel", "fedora"]:
-            pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
-        elif dist.name == "SuSE":
+        elif self.dist.name in ["centos", "rhel", "fedora"]:
+            if (self.dist.name == 'rhel' and self.dist.version == '9'):
+                self.log.info("libhugetlbfs is not available RHEL 9.x onwards,\
+                                so tests related to hugepage will be cancelled")
+                pkgs.extend(['numactl-devel'])
+            else:
+                pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
```

Comment:
Adjust this accordingly

Reply:
58         elif self.dist.name in ["centos", "rhel", "fedora"]:
 59             if (self.dist.name == 'rhel' and self.dist.version == '9'):
 60                 self.log.info("hugepage tests would be cancelled due to \
 61                                 non-availability of libhugetlbfs in RHEL-9")
 62             else:
 63                 #pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
 64                 exp_cmd = "export HAVE_HUGETLB_HEADER"
 65                 process.system(exp_cmd, shell=True, sudo=True,\
 66                         ignore_status=True)
 67             pkgs.extend(['numactl-devel'])
**I assume these changes should be fine !**